common/types.ts

```diff
@@ -13,3 +14,25 @@ export interface FetchDataFrameContext {
 }
 
 export type FetchFunction<T, P = void> = (params?: P) => Observable<T>;
+
+// ref: https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/job-states.html
+export enum SparkJobState {
+  SUBMITTED,
```

Comment:
i think this needs to be set to a string like
```ts
SUBMITTED = 'SUBMITTED',
PENDING = 'PENDING',
...
```

Sauce: https://www.typescriptlang.org/docs/handbook/enums.html#string-enums